tests/test_packages/test_skills/test_generic_buyer/test_behaviours.py

```diff
@@ -0,0 +1,389 @@
+
+        # operation
+        search_behaviour.act()
+
+        # after
+        assert self.get_quantity_in_outbox() == 1
+        actual_message = cast(LedgerApiMessage, self.get_message_from_outbox())
+        assert type(actual_message) == OefSearchMessage
+        assert actual_message.performative == expected_performative
+        assert actual_message.to == expected_counterparty
+        assert actual_message.query == expected_query
+
+    def test_search_behaviour_act_not_is_searching(self):
+        """Test the act method of the search behaviour where is_searching is False."""
+        # setup
+        strategy = cast(GenericStrategy, self.skill.skill_context.strategy)
+        strategy._is_searching = False
+        search_behaviour = cast(
+            GenericSearchBehaviour, self.skill.skill_context.behaviours.search
+        )
+
+        # before
+        assert self.get_quantity_in_outbox() == 0
+
+        # operation
+        search_behaviour.act()
+
+        # after
+        assert self.get_quantity_in_outbox() == 0
+
+
+class TestSkillHandler(BaseSkillTestCase):
+    """Test handlers of generic buyer."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "generic_buyer")
+
+    def test_fipa_handler_handle_unidentified_dialogue(self):
+        """Test the _handle_unidentified_dialogue method of the fipa handler."""
+        # setup
+        fipa_handler = cast(GenericFipaHandler, self.skill.skill_context.handlers.fipa)
+
+        incoming_message = FipaMessage(
+            dialogue_reference=("1", "0"),
+            message_id=1,
+            target=0,
+            performative=FipaMessage.Performative.ACCEPT,
+        )
+        incoming_message.sender = "counterparty"
+        incoming_message.to = "me"
+
+        expected_performative = DefaultMessage.Performative.ERROR
+        expected_counterparty = incoming_message.sender
+        expected_error_code = DefaultMessage.ErrorCode.INVALID_DIALOGUE
+        expected_error_msg = "Invalid dialogue."
+        expected_error_data = {"fipa_message": incoming_message.encode()}
+
+        # before
+        assert self.get_quantity_in_outbox() == 0
+
+        # operation
+        fipa_handler._handle_unidentified_dialogue(incoming_message)
+
+        # after
+        assert self.get_quantity_in_outbox() == 1
+        actual_message = cast(DefaultMessage, self.get_message_from_outbox())
+        assert type(actual_message) == DefaultMessage
+        assert actual_message.performative == expected_performative
+        assert actual_message.to == expected_counterparty
+        assert actual_message.error_code == expected_error_code
+        assert actual_message.error_msg == expected_error_msg
+        assert actual_message.error_data == expected_error_data
+
+    def test_fipa_handler_handle_propose(self):
+        """Test the _handle_propose method of the fipa handler."""
+        # ToDo need to mock affordable and acceptable values
```

Comment:
This is an example that shows mocking is needed. Need to be able to mock `is_acceptable_proposal` and `is_affordable_proposal`, otherwise it would be too much manual configuration to ensure those values are what you want, also making the test not independent of the skill's configuration yaml.